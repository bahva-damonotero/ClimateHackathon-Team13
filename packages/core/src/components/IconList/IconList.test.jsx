import React from 'react';
import renderer from 'react-test-renderer';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCode, faImage, faLink } from '@fortawesome/free-solid-svg-icons';
import spriteSvg from '@uswds/uswds/img/sprite.svg';

import { IconList, IconListContent, IconListIcon, IconListItem } from '.';

describe('IconList', () => {
  it('should match snapshot (default)', () => {
    const tree = renderer
      .create(
        <IconList>
          <IconListItem>
            <IconListIcon
              color="green"
              sprite={spriteSvg}
              identifier="check_circle"
            />
            <IconListContent>
              Wash your hands for 20 seconds with soap
            </IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon
              color="green"
              sprite={spriteSvg}
              identifier="check_circle"
            />
            <IconListContent>Stay six feet away from others</IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon color="red" sprite={spriteSvg} identifier="cancel" />
            <IconListContent>Avoid large gatherings</IconListContent>
          </IconListItem>
        </IconList>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot (color)', () => {
    const tree = renderer
      .create(
        <IconList color="primary">
          <IconListItem>
            <IconListIcon sprite={spriteSvg} identifier="thumb_up_alt" />
            <IconListContent>No processing lines</IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon sprite={spriteSvg} identifier="thumb_up_alt" />
            <IconListContent>
              Access to expedited entry benefits in other countries
            </IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon sprite={spriteSvg} identifier="thumb_up_alt" />
            <IconListContent>Available at major U.S. airports</IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon sprite={spriteSvg} identifier="thumb_up_alt" />
            <IconListContent>Reduced wait times</IconListContent>
          </IconListItem>
        </IconList>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot (custom size)', () => {
    const tree = renderer
      .create(
        <IconList size="lg">
          <IconListItem>
            <IconListIcon color="blue" sprite={spriteSvg} identifier="help" />
            <IconListContent>
              <span className="text-bold">Timing.</span> Is now the right time
              to start a business?
            </IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon color="blue" sprite={spriteSvg} identifier="help" />
            <IconListContent>
              <span className="text-bold">Funding.</span> Do I have enough money
              to launch a business?
            </IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon color="blue" sprite={spriteSvg} identifier="help" />
            <IconListContent>
              <span className="text-bold">Need.</span> Will this business fill a
              real need for my customers?
            </IconListContent>
          </IconListItem>
        </IconList>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot (custom icons)', () => {
    const tree = renderer
      .create(
        <IconList>
          <IconListItem>
            <IconListIcon color="red">
              <FontAwesomeIcon className="usa-icon" icon={faCode} />
            </IconListIcon>
            <IconListContent>
              This item uses a Font Awesome icon.
            </IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon color="green">
              <FontAwesomeIcon className="usa-icon" icon={faImage} />
            </IconListIcon>
            <IconListContent>
              This item uses a Font Awesome icon.
            </IconListContent>
          </IconListItem>

          <IconListItem>
            <IconListIcon color="blue">
              <FontAwesomeIcon className="usa-icon" icon={faLink} />
            </IconListIcon>
            <IconListContent>
              This item uses a Font Awesome icon.
            </IconListContent>
          </IconListItem>
        </IconList>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
