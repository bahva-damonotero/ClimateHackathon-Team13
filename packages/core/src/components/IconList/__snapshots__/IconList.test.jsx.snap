// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconList should match snapshot (color) 1`] = `
<ul
  className="usa-icon-list usa-icon-list--primary"
>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#thumb_up_alt"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      No processing lines
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#thumb_up_alt"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      Access to expedited entry benefits in other countries
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#thumb_up_alt"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      Available at major U.S. airports
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#thumb_up_alt"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      Reduced wait times
    </div>
  </li>
</ul>
`;

exports[`IconList should match snapshot (custom icons) 1`] = `
<ul
  className="usa-icon-list"
>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-red"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-code fa-w-20 usa-icon"
        data-icon="code"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.9 511.5l-61-17.7c-6.4-1.8-10-8.5-8.2-14.9L346.2 8.7c1.8-6.4 8.5-10 14.9-8.2l61 17.7c6.4 1.8 10 8.5 8.2 14.9L293.8 503.3c-1.9 6.4-8.5 10.1-14.9 8.2zm-114-112.2l43.5-46.4c4.6-4.9 4.3-12.7-.8-17.2L117 256l90.6-79.7c5.1-4.5 5.5-12.3.8-17.2l-43.5-46.4c-4.5-4.8-12.1-5.1-17-.5L3.8 247.2c-5.1 4.7-5.1 12.8 0 17.5l144.1 135.1c4.9 4.6 12.5 4.4 17-.5zm327.2.6l144.1-135.1c5.1-4.7 5.1-12.8 0-17.5L492.1 112.1c-4.8-4.5-12.4-4.3-17 .5L431.6 159c-4.6 4.9-4.3 12.7.8 17.2L523 256l-90.6 79.7c-5.1 4.5-5.5 12.3-.8 17.2l43.5 46.4c4.5 4.9 12.1 5.1 17 .6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      This item uses a Font Awesome icon.
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-green"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-image fa-w-16 usa-icon"
        data-icon="image"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 448H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h416c26.51 0 48 21.49 48 48v288c0 26.51-21.49 48-48 48zM112 120c-30.928 0-56 25.072-56 56s25.072 56 56 56 56-25.072 56-56-25.072-56-56-56zM64 384h384V272l-87.515-87.515c-4.686-4.686-12.284-4.686-16.971 0L208 320l-55.515-55.515c-4.686-4.686-12.284-4.686-16.971 0L64 336v48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      This item uses a Font Awesome icon.
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-blue"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-link fa-w-16 usa-icon"
        data-icon="link"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      This item uses a Font Awesome icon.
    </div>
  </li>
</ul>
`;

exports[`IconList should match snapshot (custom size) 1`] = `
<ul
  className="usa-icon-list usa-icon-list--size-lg"
>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-blue"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#help"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      <span
        className="text-bold"
      >
        Timing.
      </span>
       Is now the right time to start a business?
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-blue"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#help"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      <span
        className="text-bold"
      >
        Funding.
      </span>
       Do I have enough money to launch a business?
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-blue"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#help"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      <span
        className="text-bold"
      >
        Need.
      </span>
       Will this business fill a real need for my customers?
    </div>
  </li>
</ul>
`;

exports[`IconList should match snapshot (default) 1`] = `
<ul
  className="usa-icon-list"
>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-green"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#check_circle"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      Wash your hands for 20 seconds with soap
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-green"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#check_circle"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      Stay six feet away from others
    </div>
  </li>
  <li
    className="usa-icon-list__item"
  >
    <div
      aria-hidden="true"
      className="usa-icon-list__icon text-red"
    >
      <svg
        className="usa-icon"
        role="img"
      >
        <use
          xlinkHref="test-file-stub#cancel"
        />
      </svg>
    </div>
    <div
      className="usa-icon-list__content"
    >
      Avoid large gatherings
    </div>
  </li>
</ul>
`;
